src/hb/ot_shaper_vowel_constraints.rs

```diff
@@ -365,14 +365,13 @@ pub fn preprocess_text_vowel_constraints(buffer: &mut hb_buffer_t) {
             while buffer.idx + 1 < buffer.len {
                 #[allow(unused_mut)]
                 let mut matched = false;
-                match buffer.cur(0).glyph_id {
-                    0x112B0 => match buffer.cur(1).glyph_id {
+                if buffer.cur(0).glyph_id == 0x112B0 { 
+                    match buffer.cur(1).glyph_id {
```

Comment:
This is a generated file, so this will be overwritten the next time this file is re-generated.

Reply:
I guess the python script could add the lint somewhere at the top of the file? 

```py
print('#![allow(clippy::single_match)]')
```